python/xorq/flight/__init__.py

```diff
@@ -130,10 +132,23 @@ def __init__(
         self.flight_url = flight_url or FlightUrl()
         self.certificate_path = certificate_path
         self.key_path = key_path
+        self.cleanup = False
+
+        if enable_tls and (certificate_path is None and key_path is None):
+            from xorq.common.utils.tls_utils import create_tls_keypair
+
+            with tempfile.TemporaryDirectory() as tls_dir:
+                tls_dir = Path(tls_dir).resolve()
+
+            self.certificate_path = str(tls_dir / "server.crt")
+            self.key_path = str(tls_dir / "server.key")
+            create_tls_keypair(self.certificate_path, self.key_path)
+            self.cleanup = True
+
         self.root_certificates = root_certificates
         self.auth = auth
         self.connection = connection
-        self.verify_client = verify_client
+        self.verify_client = False
```

Comment:
@mesejo why is `verify_client` always false?

Reply:
we want to be able to do something like this
https://github.com/apache/arrow/blob/apache-arrow-20.0.0/python/pyarrow/tests/test_flight.py#L2019-L2035

Reply:
I'm solving it in https://github.com/xorq-labs/xorq/pull/958